C# 8 Using Statement produces compile error

### What You Are Seeing?
When using the new C#8 `using var...` statement, the cake script produces compile errors.

### What is Expected?
This script should compile successfully.

### What version of Cake are you using?
0.35.0

### Are you running on a 32 or 64 bit system?
64bit

### What environment are you running on?  Windows? Linux? Mac?
Windows 10

### Are you running on a CI Server?  If so, which one?
No

### How Did You Get This To Happen? (Steps to Reproduce)

```
Task("C#8-Using-Statement")
    .Does(() =>
    {
        using var file = System.IO.File.Create(@".\temp.txt");
    });
```

### Output Log
```
> .\build.ps1 -target C#8-Using-Statement

Preparing to run build script...
Running build script...
(5,11): error CS1002: ; expected
(5,7): error CS0825: The contextual keyword 'var' may only appear within a local variable declaration or in script code
(5,11): error CS0103: The name 'file' does not exist in the current context
```

--- a/src/Cake.Core/Scripting/Processors/UsingStatementProcessor.cs
+++ b/src/Cake.Core/Scripting/Processors/UsingStatementProcessor.cs
@@ -30,9 +30,19 @@ namespace Cake.Core.Scripting.Processors
                 return false;
             }
 
-            // Using block?
+            // Using disposable block?
             var @namespace = tokens[1].TrimEnd(';');
             if (@namespace.StartsWith("("))
+            {
+                return false;
+            }
+
+            // Using disposable statement?
+            const int usingLength = 5;
+            int openParentheses = line.IndexOf('(', usingLength),
+                closeParentheses = openParentheses < usingLength ? -1 : line.IndexOf(')', openParentheses);
+
+            if (closeParentheses > openParentheses)
             {
                 return false;
             }

--- a/tests/integration/Cake.Core/Scripting/UsingDirective.cake
+++ b/tests/integration/Cake.Core/Scripting/UsingDirective.cake
@@ -6,7 +6,17 @@ Task("Cake.Core.Scripting.UsingDirective.UsingStatic")
     Information(Round(1.1));
 });
 
+Task("Cake.Core.Scripting.UsingDirective.UsingDisposable")
+    .Does(() =>
+{
+    // Intentionally not indented
+using var @string = new StringReader("String");
+    using var reader = new StringReader("Reader");
+    Information("{0}{1}", @string.ReadLine(), reader.ReadLine());
+});
+
 //////////////////////////////////////////////////////////////////////////////
 
 Task("Cake.Core.Scripting.UsingDirective")
-    .IsDependentOn("Cake.Core.Scripting.UsingDirective.UsingStatic");
+    .IsDependentOn("Cake.Core.Scripting.UsingDirective.UsingStatic")
+    .IsDependentOn("Cake.Core.Scripting.UsingDirective.UsingDisposable");

